src/Adapter/Profile/Permission/QueryHandler/GetPermissionsForConfigurationHandler.php

```diff
@@ -0,0 +1,304 @@
+use Module;
+use PrestaShop\PrestaShop\Core\Domain\Profile\Permission\Query\GetPermissionsForConfiguration;
+use PrestaShop\PrestaShop\Core\Domain\Profile\Permission\QueryHandler\GetPermissionsForConfigurationHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Profile\Permission\QueryResult\ConfigurablePermissions;
+use PrestaShopBundle\Security\Voter\PageVoter;
+use Profile;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Tab;
+
+/**
+ * Get configuratble permissions
+ *
+ * @internal
+ */
+final class GetPermissionsForConfigurationHandler implements GetPermissionsForConfigurationHandlerInterface
+{
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $authorizationChecker;
+
+    /**
+     * @param AuthorizationCheckerInterface $authorizationChecker
+     */
+    public function __construct(AuthorizationCheckerInterface $authorizationChecker)
+    {
+        $this->authorizationChecker = $authorizationChecker;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetPermissionsForConfiguration $query)
+    {
+        $profiles = $this->getProfilesForPermissionsConfiguration();
+        $tabs = $this->getTabsForPermissionsConfiguration();
+        $permissions = ['view', 'add', 'edit', 'delete'];
+
+        $tabPermissionsForProfiles = $this->getTabPermissionsForProfiles($profiles);
+        $modulePermissionsForProfiles = $this->getModulePermissionsForProfiles($profiles);
+
+        $permissionIds = ['view' => 0, 'add' => 1, 'edit' => 2, 'delete' => 3, 'all' => 4];
+        $employeeProfileId = $query->getEmployeeProfileId()->getValue();
+        $canEmployeeEditPermissions = $this->authorizationChecker->isGranted(PageVoter::UPDATE, 'AdminAccess');
+
+        $bulkConfigurationPermissions = $this->getBulkConfigurationForProfiles(
+            $employeeProfileId,
+            $canEmployeeEditPermissions,
+            $tabPermissionsForProfiles,
+            $profiles,
+            $tabs,
+            $permissions
+        );
+
+        return new ConfigurablePermissions(
+            $tabPermissionsForProfiles,
+            $modulePermissionsForProfiles,
+            $profiles,
+            $tabs,
+            $bulkConfigurationPermissions,
+            $permissions,
+            $permissionIds,
+            $employeeProfileId,
+            $canEmployeeEditPermissions
+        );
+    }
+
+    /**
+     * @return int[] IDs of non configurable tabs
+     */
+    private function getNonConfigurableTabs()
+    {
+        return [
+            Tab::getIdFromClassName('AdminLogin'),
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    private function getProfilesForPermissionsConfiguration()
+    {
+        $legacyProfiles = Profile::getProfiles(Context::getContext()->language->id);
+        $profiles = [];
+
+        foreach ($legacyProfiles as $profile) {
+            $isAdministrator = (int) $profile['id_profile'] === _PS_ADMIN_PROFILE_;
```

Comment:
I think I saw in a PR dealing with multishop that `_PS_ADMIN_PROFILE_` can become an array in a multishop context 🤔 not sure though

Reply:
I see it's used in codebase as `int`.